src/components/NestedAccordionMenu/MenuItems/index.js

```diff
@@ -0,0 +1,98 @@
+import React, { Fragment, useState } from "react";
+import List from "@material-ui/core/List";
+import ListItem from "@material-ui/core/ListItem";
+import ExpandMoreIcon from "@material-ui/icons/ExpandMore";
+import ExpandLessIcon from "@material-ui/icons/ExpandLess";
+import Collapse from "@material-ui/core/Collapse";
+import ListItemIcon from "@material-ui/core/ListItemIcon";
+import ListItemText from "@material-ui/core/ListItemText";
+import useStyles from "./styles";
+function MenuItem({ depth = 0, expanded, item, currentPath, ...rest }) {
+  const classes = useStyles({ depth: depth });
+  const [collapsed, setCollapsed] = useState(true);
+  const [highlightcolor, setHighlightcolor] = useState(false);
+  const { route, name, items, Icon } = item;
+  const mainroute = rest.mainroute;
+  let highlightclass = "itemNohighlighted";
+  function toggleCollapse() {
+    setCollapsed(prevValue => !prevValue);
+  }
+
+  function onClick(e) {
+    if (Array.isArray(items)) {
+      toggleCollapse();
```

Comment:
What if it’s not an array? How will it route?  Imho they are 2 solutions:
- Add a Link from react-router to Menu item if it has no items
- bubble up the event to a func prop on the main component. Something like onChange